ie/apply-action.go

```diff
@@ -47,18 +46,60 @@ func (i *IE) ApplyAction() ([]byte, error) {
 	}
 }
 
+// ValidateApplyAction can be used to facilitate the detection of some inconsistencies in Apply Action flags.
+// Its use is optional because validation could also be done on upper layers, or completely skipped for testing purposes.
+func (i *IE) ValidateApplyAction() error {
+	if i.Type != ApplyAction {
+		return &InvalidTypeError{Type: i.Type}
+	}
+	// One and only one of the DROP, FORW, BUFF, IPMA and IPMD flags shall be set to "1".
+	flags := []bool{i.HasDROP(), i.HasFORW(), i.HasBUFF(), i.HasIPMA(), i.HasIPMD()}
+	counter := 0
+	for _, v := range flags {
+		if v {
+			counter++
+		}
+	}
+	if counter != 1 {
+		return ErrMalformed
+	}
+	// The NOCP flag and BDPN flag may only be set if the BUFF flag is set.
+	if (i.HasNOCP() || i.HasBDPN()) && !i.HasBUFF() {
+		return ErrMalformed
+	}
+	// The DUPL flag may be set with any of the DROP, FORW, BUFF and NOCP flags.
+	if i.HasDUPL() && !(i.HasDROP() || i.HasFORW() || i.HasBUFF() || i.HasNOCP()) {
+		return ErrMalformed
+	}
+	// The DFRT flag may only be set if the FORW flag is set.
+	// Note: in TS 29.244 V18.0.1 (most recent as of writing), there is a typo and DFRN is stated instead of DFRT
+	if i.HasDFRT() && !i.HasFORW() {
+		return ErrMalformed
+	}
+	// The DDPN flag may be set with any of the DROP and BUFF flags.
+	if i.HasDDPN() && !(i.HasDROP() || i.HasBUFF()) {
+		return ErrMalformed
+	}
+
+	// Note: The following is also stated in TS 29.244 section 8.2.26 V18.0.1,
+	// but since "may" is used and not "may only"
+	// it cannot be used to check for inconsistent IEs:
+	// - The EDRT flag may be set if the FORW flag is set.
+	// - Both the MBSU flag and the FSSM flag may be set […]
+	return nil
+}
+
 // HasDROP reports whether an IE has DROP bit.
 func (i *IE) HasDROP() bool {
-	if len(i.Payload) < 1 {
-		return false
+	v, err := i.ApplyAction()
```

Comment:
How about rewrite like as below:
```
	switch i.Type {
	case CreateFAR, UpdateFAR, ApplyAction:
		v, err := i.ApplyAction()
		if err != nil {
			return false
		}
		return has1stBit(v[0])
	case DownlinkDataReport, DataStatus:
		v, err := i.DataStatus()
		if err != nil {
			return false
		}
		return has1stBit(v)
	default:
		return false
	}
```

Reply:
I sent another  PR(https://github.com/wmnsk/go-pfcp/pull/115) to enhance ReportingTriggers helper functions usage.
FYI.

Reply:
I removed the related commit from this branch.